runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/resources/PipelineResourcesTest.java

```diff
@@ -102,8 +100,57 @@ public void testPackagingDirectoryResourceToJarFile() throws IOException {
 
   @Test
   public void testIfThrowsWhenThereIsNoTemporaryFolderForJars() throws IOException {
-    List<String> filesToStage = new ArrayList<>();
-    filesToStage.add(tmpFolder.newFolder().getAbsolutePath());
+    List<String> filesToStage = Arrays.asList(tmpFolder.newFolder().getAbsolutePath());
+
+    IllegalArgumentException exception =
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> PipelineResources.prepareFilesForStaging(filesToStage, null));
```

Comment:
I guess we don't have nullness checks set up for this package, otherwise we could statically guarantee this can't be null.